billy-core-jpa/src/main/java/com/premiumminds/billy/persistence/entities/jpa/JPAGenericInvoiceEntryEntity.java

```diff
@@ -397,4 +406,9 @@ public void setAmountType(AmountType type) {
         this.type = type;
     }
 
+    @Override
+    public void setExternalID(final ExternalID<GenericInvoiceEntry> externalID) {
+        this.externalID = externalID.toString();
```

Comment:
I'd recommend staying away from using `.toString()` to obtain the `String` internal representation on any kind of wrapper or actually any class in general. 
As `.toString()` is implemented by any object and might not be implemented/overridden as needed on a given class there's a high risk of the return value not being the needed or expected. Also, since it is implemented every `Object` in the language, any relevant change on the class being called or change of type (e.g. for an invalid type for case) will not be caught by the compiler and potential bug might easily arise.